src/java.base/share/classes/sun/security/ssl/X509KeyManagerImpl.java

```diff
@@ -77,15 +79,63 @@ final class X509KeyManagerImpl extends X509ExtendedKeyManager
     // cached entries
     private final Map<String,Reference<PrivateKeyEntry>> entryCacheMap;
 
+    private boolean ksP12;
+    /*
+     * The credentials from the PKCS12KeyStore as
+     * Map: String(builderIndex.entryAlias) -> X509Credentials(credentials)
+     */
+    private final Map<String, X509Credentials> credentialsMap;
+
+    private static class X509Credentials {
+        final PrivateKey privateKey;
+        final X509Certificate[] certificates;
+
+        X509Credentials(PrivateKey privateKey, X509Certificate[] certificates) {
+            this.privateKey = privateKey;
+            this.certificates = certificates;
+        }
+    }
+
     X509KeyManagerImpl(Builder builder) {
         this(Collections.singletonList(builder));
     }
 
     X509KeyManagerImpl(List<Builder> builders) {
         this.builders = builders;
         uidCounter = new AtomicLong();
-        entryCacheMap = Collections.synchronizedMap
-                        (new SizedMap<>());
+        KeyStore keyStore = null;
```

Comment:
It may be better to define `keyStore` in the below for-loop.

Reply:
Done.